packages/terminal/src/browser/terminal-linkmatcher-files.ts

```diff
@@ -0,0 +1,172 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { ApplicationServer } from '@theia/core/lib/common/application-protocol';
+import { OS } from '@theia/core/lib/common';
+import { OpenerService } from '@theia/core/lib/browser';
+import { Position } from '@theia/editor/lib/browser';
+import { AbstractCmdClickTerminalContribution } from './terminal-linkmatcher';
+import { TerminalWidgetImpl } from './terminal-widget-impl';
+import { Path } from '@theia/core';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class TerminalLinkmatcherFiles extends AbstractCmdClickTerminalContribution {
+
+    protected backendOs: Promise<OS.Type>;
+
+    constructor(
+        @inject(ApplicationServer) appServer: ApplicationServer,
+        @inject(OpenerService) protected openerService: OpenerService,
+        @inject(FileSystem) protected fileSystem: FileSystem,
+    ) {
+        super();
+        this.backendOs = appServer.getBackendOS();
+    }
+
+    async getRegExp(): Promise<RegExp> {
+        const os = await this.backendOs;
+        const baseLocalLinkClause = os === OS.Type.Windows ? winLocalLinkClause : unixLocalLinkClause;
+        return new RegExp(`${baseLocalLinkClause}(${lineAndColumnClause})`);
+    }
+
+    getValidate(terminalWidget: TerminalWidgetImpl): (link: string) => Promise<boolean> {
+        return async match => {
+            const toOpen = await this.toURI(match, await terminalWidget.cwd);
```

Comment:
example:
```
<   "//": [
```
`//` is matched as URI by the regex, but cannot be validated as URI internally.

Reply:
matched by this:
```
/(((\.\.?|\~)|([^\0\s!$`&*()\[\]+'":;\\])+)?(\/([^\0\s!$`&*()\[\]+'":;\\])+)+)(((\S*)",[  ]line[  ]((\d+)([  ]column[  ](\d+))?))|((\S*)[  ]on[  ]line[  ]((\d+)(,[  ]column[  ](\d+))?))|((\S*):line[  ]((\d+)(,[  ]column[  ](\d+))?))|(([^\s\(\)]*)(\s?[\(\[](\d+)(,\s?(\d+))?)[\)\]])|(([^:\s\(\)<>'"\[\]]*)(:(\d+))?(:(\d+))?))/g
```

Reply:
What does this mean, @AlexTugarev?

Reply:
I've seen Theia URI validation errors as result of having the string `//` somewhere in the terminal. Turned out, that it's matched by [`winExcludedPathCharactersClause `](https://github.com/theia-ide/theia/pull/5157/files#diff-28a6b2435812cc9d26163a55fb58d302R153) and passed to `getValidate`. Down the road there is some URI construction and `//` doesn't pass the validation. Sven added a try-catch here.

Reply:
👍 Thanks! Meanwhile, I have noticed the `Outdated` marker on the comment.